app/controllers/facility_orders_controller.rb

```diff
@@ -32,8 +32,13 @@ def initialize
   def show
     @order_details = @order.order_details.ordered_by_parents
     @order_details = @order_details.includes(:reservation, :order_status, :product, :order)
+    updated_facility_id = params[:updated_facility_id]
     if params[:refresh]
-      render partial: "order_table", locals: { order_details: @order_details, cross_core: false }
+      if updated_facility_id == current_facility.id.to_s
+        render partial: "order_table", locals: { order_details: @order_details, cross_core: false, facility: current_facility }
+      else
+        render partial: "order_table", locals: { order_details: @cross_core_data_by_facility_id[updated_facility_id][:order_details], cross_core: true, facility: updated_facility_id }
```

Comment:
I think this would be more readable if you put the order detail into a variable:
```
cross_core_order_details = @cross_core_data_by_facility_id[updated_facility_id][:order_details]
```